### DISCUSSION TITLE
Undefined symbols trying to build with nix (mainly OS X)

### DISCUSSION THREAD

**Discussion by spease** [halide/Halide#5367] [Q&A] [answered]
Example:
```
Undefined symbols for architecture x86_64:
  "_LLVMInitializeAArch64AsmParser", referenced from:
      void std::__1::__call_once_proxy<std::__1::tuple<Halide::Internal::CodeGen_LLVM::initialize_llvm()::$_0&&> >(void*) in libHalide.a(CodeGen_LLVM.cpp.o)
```

Linux also produces a slew of undefined reference errors, but these seem to be much more generic. I am including llvmPackages.llvm as either a buildInput or propagatedBuildInput (makes no difference).

Thanks.

**Comment by steven-johnson** [comment]
First guess is that you have an LLVM that was built without AArch64 support. Did you build it from source? If so, specify something like `-D LLVM_TARGETS_TO_BUILD="X86;ARM;NVPTX;AArch64;Mips;PowerPC;Hexagon;WebAssembly"` when configuring CMake.

**Comment by spease** [comment]
Nonconclusively, it looks like it’s probably building with all architectures since that’s the LLVM default.

Another thing I’ve noticed is that the official release at least for 8.0.0, comes with a shared and static library. The cmake API seems to make use of both depending on which function is called?. However if I build the library, it only builds one toggleable with a variable.

What should I be doing to build a package for a package system? Like which make target should I be using, or what script should I be following?

**Comment by steven-johnson** [comment, marked as answer]
The 8.0 release is pretty old at this point -- TBH you'd be better off upgrading to our release 10.

**Reply by spease** [reply]
This is what ultimately solved the build issues I was having
